api/rhsm-subscriptions-api-v1-spec.yaml

```diff
@@ -140,7 +140,7 @@ paths:
       - name: category
         in: query
         schema:
-          $ref: '#/components/schemas/ReportCategory'
```

Comment:
I initially did this, but found that we had to duplicate a lot of logic just because these types are now v1 and v2. 
Anyway, it makes sense to me. I will re-evaluate this and we can check if it's worthy or not.

Reply:
Update: after taking over the review, I now recalled why I had to make the model common: we have some entities that directly use the model API. So, we can't map these entities to multiple versioned model APIs. 
Example: https://github.com/RedHatInsights/rhsm-subscriptions/blob/main/swatch-core/src/main/java/org/candlepin/subscriptions/db/model/Usage.java

Reply:
Resolved in latest commit